recipe/build-python.sh

```diff
@@ -1,26 +1,32 @@
 #!/bin/sh
 set -ex
 
+# clean up remnants from cpp build phase
+rm -rf build
+mkdir build
+cd build
+
 cmake -G Ninja \
-      ${CMAKE_ARGS} \
-      -DCMAKE_BUILD_TYPE=Release \
-      -DCMAKE_CXX_STANDARD=17 \
-      -DCMAKE_INSTALL_LIBDIR=lib \
-      -DCMAKE_INSTALL_PREFIX=$PREFIX \
-      -DBUILD_SHARED_LIBS=ON \
-      -DBUILD_DEPS=OFF \
-      -DUSE_SCIP=OFF \
-      -S. \
-      -Bbuild \
-      -DBUILD_SAMPLES=OFF \
-      -DBUILD_EXAMPLES=OFF \
-      -DBUILD_PYTHON=ON \
-      -DBUILD_pybind11=OFF \
-      -DFETCH_PYTHON_DEPS=OFF \
-      -DBUILD_TESTING=OFF \
-      -DPython3_EXECUTABLE="$PYTHON"
+    ${CMAKE_ARGS} \
+    -DCMAKE_BUILD_TYPE=Release \
+    -DCMAKE_CXX_STANDARD=17 \
+    -DCMAKE_INSTALL_LIBDIR=lib \
+    -DCMAKE_INSTALL_PREFIX=$PREFIX \
+    -DBUILD_SHARED_LIBS=ON \
+    -DBUILD_CXX=OFF \
```

Comment:
`BUILD_CXX=OFF` this is used to build libglop.so (only useful/needed by SCIP to build lpiglop iirc)
since Python,java,.Net depend on c++ (just swig wrapper) you need it to ON to be able to build the python layer

Reply:
Ah, I hadn't seen comment yet when I wrote the following [question](https://github.com/conda-forge/or-tools-feedstock/issues/2#issuecomment-1567817671) in the bindings issue:
> > * [x]   Google GLOP (can't be disable IIRC)
> 
> @Mizux, I'm looking at enabling more solvers (in addition to scip), and I noted that, by default, GLOP does _not_ get [built](https://github.com/google/or-tools/blob/v9.6/CMakeLists.txt#L119) when `BUILD_CPP=ON`. Is that intentional? The last commit [touching](https://github.com/google/or-tools/commit/40dc9ca87f0e63d03203121940ceeb7656a3d8b7) that line only says "Update glop management", but it was OFF before that as well... 🤔

I don't see how `BUILD_CXX=ON` is necessary TBH (well, to be fair, I'm patching out a hard error for that already 😅), in particular, all the python things bits seem to build an link correctly.

I've moved glop to the C++ side for now, that seems to make more sense to me (but happy to be corrected!).

Reply:
Ah, it seems that when both `BUILD_CPP=ON` and `BUILD_GLOP=ON` are specified, glop gets ignored:
```
Build standalone Glop: OFF
```

I can patch this as well of course, but I'd like to understand this separation? Whats wrong about having both `libortools.so` and `libglop.so` at the same time?

Reply:
OK nevermind, I see this doesn't work:
```
CMake Error at cmake/glop.cmake:44 (add_custom_command):
  Attempt to add a custom rule to output

    /home/conda/feedstock_root/build_artifacts/or-tools-package_1685431583287/work/build/ortools/linear_solver/linear_solver.pb.cc.rule

  which already has a custom rule.
```

Not sure what relevance GLOP then has to the packaging here, if it cannot be build (and installed) with `libortools`

Reply:
> Thanks for the quick feedback!
> 
> > So it's really a hacky truncated version of libortools only containing GLOP stuff
> 
> But the regular `libortools` _does not_ contain GLOP? I'm working my way through the various PRs, but scip worked fine with `USE_SCIP=ON`, and obviously GLOP could not have been built because `BUILD_CXX=ON` was also the case.

No libortools contains glop too. aka libglop is just a subset of libortools.so

> 
> > BUILD_CXX is mandatory to build python package because we don't support (yet ?) to be able to depends on system installed ortools package
> 
> This is what `0001-depend-on-libortools-for-python-bindings.patch` in this PR enables. I'm willing to patch this, because I _really_ don't want to rebuild the C++ lib when I build python. There's more places where `.libs` could be replaced (the expected location of `libortools.so`), but the right builds already get picked up from the path automatically, after `.libs` is empty. The following is an excerpt from a previously passing CI:
> 
> ```
>    INFO (ortools-python,lib/python3.11/site-packages/ortools/pdlp/python/pywrap_pdlp.cpython-311-x86_64-linux-gnu.so): Needed DSO lib/libortools.so.9 found in home/conda/feedstock_root/build_artifacts::libortools-9.6-hcb37b0f_2
>    INFO (ortools-python,lib/python3.11/site-packages/ortools/pdlp/python/pywrap_pdlp.cpython-311-x86_64-linux-gnu.so): Needed DSO lib/libabsl_status.so.2301.0.0 found in conda-forge::libabseil-20230125.2-cxx17_h59595ed_2
>    INFO (ortools-python,lib/python3.11/site-packages/ortools/pdlp/python/pywrap_pdlp.cpython-311-x86_64-linux-gnu.so): Needed DSO lib/libprotobuf.so.32 found in conda-forge::libprotobuf-3.21.12-h3eb15da_0
>    INFO (ortools-python,lib/python3.11/site-packages/ortools/pdlp/python/pywrap_pdlp.cpython-311-x86_64-linux-gnu.so): Needed DSO lib/libstdc++.so.6 found in conda-forge::libstdcxx-ng-12.2.0-h46fd767_19
>    INFO (ortools-python,lib/python3.11/site-packages/ortools/pdlp/python/pywrap_pdlp.cpython-311-x86_64-linux-gnu.so): Needed DSO lib/libgcc_s.so.1 found in conda-forge::libgcc-ng-12.2.0-h65d4601_19
>    INFO (ortools-python,lib/python3.11/site-packages/ortools/pdlp/python/pywrap_pdlp.cpython-311-x86_64-linux-gnu.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge::sysroot_linux-64-2.12-he073ed8_15
>    INFO (ortools-python,lib/python3.11/site-packages/ortools/algorithms/_pywrapknapsack_solver.so): Needed DSO lib/libortools.so.9 found in home/conda/feedstock_root/build_artifacts::libortools-9.6-hcb37b0f_2
>    INFO (ortools-python,lib/python3.11/site-packages/ortools/algorithms/_pywrapknapsack_solver.so): Needed DSO lib/libabsl_flags_internal.so.2301.0.0 found in conda-forge::libabseil-20230125.2-cxx17_h59595ed_2
>    INFO (ortools-python,lib/python3.11/site-packages/ortools/algorithms/_pywrapknapsack_solver.so): Needed DSO lib/libabsl_time.so.2301.0.0 found in conda-forge::libabseil-20230125.2-cxx17_h59595ed_2
>    INFO (ortools-python,lib/python3.11/site-packages/ortools/algorithms/_pywrapknapsack_solver.so): Needed DSO lib/libstdc++.so.6 found in conda-forge::libstdcxx-ng-12.2.0-h46fd767_19
>    INFO (ortools-python,lib/python3.11/site-packages/ortools/algorithms/_pywrapknapsack_solver.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libm.so.6 found in CDT/compiler package conda-forge::sysroot_linux-64-2.12-he073ed8_15
>    INFO (ortools-python,lib/python3.11/site-packages/ortools/algorithms/_pywrapknapsack_solver.so): Needed DSO lib/libgcc_s.so.1 found in conda-forge::libgcc-ng-12.2.0-h65d4601_19
>    INFO (ortools-python,lib/python3.11/site-packages/ortools/algorithms/_pywrapknapsack_solver.so): Needed DSO x86_64-conda-linux-gnu/sysroot/lib64/libc.so.6 found in CDT/compiler package conda-forge::sysroot_linux-64-2.12-he073ed8_15
>   [...]
> ```

Reply:
@Mizux, please let me know if you have any other comments here, I'd like to merge this soon.

Reply:
> Any more comments or would you be fine with the approach here for now (despite the patching)?
> 
> I've even extended the python-tests to import all the modules that are somehow exposed to compiled artefacts, and they all pass.
> 
> If you have other ideas what we should test (e.g. run the full python test suite), I'm happy to try that.
> 
> That said, I want move forward a bit with: this PR -> cross-compilation -> enable scip -> migrate protobuf. upside_down_face

to move forward, remaining issue on our side:
* cross-compilation: add `BUILD_HOST_TOOLS`  option so you can optout and provide your own protoc
  * add test workflow (adding option to tools/cross-compile.sh :thinking: )

* officially allow to build python without building libortools .so
  * need to fix setup.py
  * change the guard in python.cmake
  * add a test job to test this scenario (note: close to https://github.com/google/or-tools/blob/main/.github/workflows/amd64_linux_cmake_system_deps.yml)